File: CP_Sheet_PH/Loops/O_Fibonacci.cpp
/*
________________________
|                      |
|     AB Atiq          |
|______________________|
*/

// Problem link: https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/O

#include <bits/stdc++.h>
using namespace std;
int main()
{
    int n;
    cin >> n;
    if (n == 1)
    {
        cout << 0 << endl;
    }
    else if (n == 2)
    {
        cout << 1 << endl;
    }
    else
    {
        long long int a = 0, b = 1, cur;
        for (int i = 3; i <= n; i++)
        {
            cur = a + b;
            a = b;
            b = cur;
        }
        cout << b << endl;
    }

    return 0;
}
/*
#include <bits/stdc++.h>
using namespace std;

int main()
{
    int n;
    cin >> n;

    vector<long long> fib = {0, 1};

    for (int i = 2; i < n; i++)
    {
        fib.push_back(fib[i - 1] + fib[i - 2]);
    }

    cout << fib[n - 1];

    return 0;
} */

File: C-5-SDT/OOP_and_Python_Programming/M2.5/O_Fibonacci.py
# https://codeforces.com/group/MWSDmqGsZm/contest/219774/problem/O

# solution 1 - using variables to store last two Fibonacci numbers
# n = int(input())
# # base cases
# if n == 1:
#     print(0)
# elif n == 2:
#     print(1)
# else:
#     a = 0   # fib(1)
#     b = 1   # fib(2)
    
#     for i in range(3, n + 1):
#         c = a + b
#         a = b
#         b = c
    
#     print(b)
    
# solution 2 - using list to store Fibonacci numbers
n = int(input())

fib = [0, 1]

for i in range(2, n):
    fib.append(fib[i-1] + fib[i-2])

print(fib[n-1])